exporter/azuremonitorexporter/factory.go

```diff
@@ -63,7 +70,8 @@ func (f *factory) createTracesExporter(
 		return nil, errUnexpectedConfigurationType
 	}
 
-	tc, errInstrumentationKeyOrConnectionString := f.getTransportChannel(exporterConfig, set.Logger)
+	f.initLogger(set.Logger)
```

Comment:
you cannot assume the same Logger is passed in for all. You might need to change this logic.

Reply:
Yes, but the `appinsights` SDK only has one global diagnostics writer. If adding each logger to listener, there'll be dup messages. I don't know any elegant way to solve this issue at this point. any suggestion?

https://github.com/microsoft/ApplicationInsights-Go/blob/c063db145320a271b46e51eef7bceb88176b82c1/appinsights/diagnostics.go#L32C1-L33C52

```go
// The one and only diagnostics writer.
var diagnosticsWriter = &diagnosticsMessageWriter{}
```